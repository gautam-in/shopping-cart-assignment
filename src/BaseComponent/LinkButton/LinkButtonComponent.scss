/******************* LinkButtonComponent *****************************/
@import '../../Styles/variable.scss';
@import '../../Styles/color-variable.scss';

$BUTTON_PADDING: 15px;

.link-button {
    // border: 1px solid $APP_BLACK;
    text-decoration: none;
    background-color: $CRIMSON_RED;
    width: 100%;
    min-height: $MEDIUM_HEIGHT;
    padding: $BUTTON_PADDING;
    font-size: $MEDIUM_FONT_SIZE;
    color: $WHITE;
    display: inline-block;
}
