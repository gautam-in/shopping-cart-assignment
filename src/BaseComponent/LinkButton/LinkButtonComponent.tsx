import { Link } from "react-router-dom";
import "./LinkButtonComponent.scss";

interface ILinkButtonComponentProps {
    id: string;
    to: string;
    buttonName: string;
    externalClassName?: string;
    onClick?: () => void;
}

export const LinkButtonComponent: React.FC<ILinkButtonComponentProps> = (
    props
) => {
    const { id, to, buttonName, externalClassName, onClick} = props;
    const BASE_CLASSNAME: string = "link-button";
    return (
        <div className={externalClassName}>
            <Link
                id={id}
                to={to}
                className={BASE_CLASSNAME}
                onClick={onClick}
            >
                {buttonName}
            </Link>
        </div>
    );
};
