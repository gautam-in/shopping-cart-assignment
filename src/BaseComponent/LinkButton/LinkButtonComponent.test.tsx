import { ShallowWrapper, shallow } from "enzyme";
import { LinkButtonComponent, ILinkButtonComponentProps } from "./LinkButtonComponent";

describe("LinkButtonComponent", () => {
    const externalClassName: string = "link-button";
    const tree: ShallowWrapper<ILinkButtonComponentProps> = shallow(
        <LinkButtonComponent id="button" buttonName="Link" to="/" externalClassName={externalClassName}/>
    );

    test("LinkButtonComponent Snapshot render", () => {
        expect(tree).toMatchSnapshot();
    });

    test("Check LinkButtonComponent has classname", () => {
        expect(tree.hasClass(externalClassName)).toEqual(true)
    });
});