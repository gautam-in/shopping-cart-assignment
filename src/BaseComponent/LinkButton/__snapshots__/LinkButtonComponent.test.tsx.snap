// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButtonComponent LinkButtonComponent Snapshot render 1`] = `ShallowWrapper {}`;
